packages/terra-toggle-section-header/src/ToggleSectionHeader.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import SectionHeader from 'terra-section-header';
+import Toggle from 'terra-toggle';
+
+const VALID_LEVELS = [1, 2, 3, 4, 5, 6];
+
+const propTypes = {
+  /**
+   * Content in the body of the toggle section header component that will be expanded or hidden.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Text to be displayed in the section header.
+   */
+  title: PropTypes.string.isRequired,
+  /**
+   * Sets the heading level. One of 1, 2, 3, 4, 5, or 6.
+   */
+  level: PropTypes.oneOf(VALID_LEVELS),
+  /**
+   * Callback function triggered when toggle-section-header is closed.
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function triggered when toggle-section-header is opened.
+   */
+  onOpen: PropTypes.func,
+  /**
+   * Used to set props and HTML attributes on the underlying section-header.
+   */
+  sectionHeaderAttrs: PropTypes.object,
+  /**
+   * Sets the toggle-section-header to be animated when it is opened or closed.
+   */
+  isAnimated: PropTypes.bool,
+  /**
+   * Sets the toggle-section-header initial state to open.
+   */
+  isInitiallyOpen: PropTypes.bool,
+};
+
+const defaultProps = {
+  level: 2,
+  sectionHeaderAttrs: {},
+};
+
+class ToggleSectionHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { isInitiallyOpen = false } = this.props;
+    this.state = { isOpen: isInitiallyOpen };
+
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isInitiallyOpen !== this.props.isInitiallyOpen) {
+      this.setState({ isOpen: nextProps.isInitiallyOpen });
+    }
+  }
+
+  handleOnClick(event) {
+    event.preventDefault();
+
+    const { onOpen, onClose } = this.props;
+
+    if (onOpen && !this.state.isOpen) {
+      onOpen();
+    } else if (onClose && this.state.isOpen) {
+      onClose();
+    }
+
+    this.setState(prevState => ({
+      isOpen: !prevState.isOpen,
+    }));
+  }
+
+  render() {
+    const {
+      children,
+      title,
+      level,
+      sectionHeaderAttrs,
+      isAnimated,
+      isInitiallyOpen,
+      ...customProps
+    } = this.props;
+
+    return (
+      <div {...customProps}>
```

Comment:
Regarding my comment above regarding `onClick` would we want to spread that customProp to this div or the section header? It seems it should apply on the section header component

Reply:
As this is supposed to be similar to `terra-toggle-button`, functionally, I followed the precedent set [here](https://github.com/cerner/terra-core/blob/toggle-section-header/packages/terra-toggle-button/src/ToggleButton.jsx#L138-L143).